platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -523,17 +526,23 @@ private void logout() {
   }
 
   private void logSecurityAuditRole() {
-    if (shouldAuditSubject()) {
-      SecurityLogger.audit(
-          "Subject with admin privileges has logged out: {}",
-          getIdpSecurityAssertion().getPrincipal().getName());
+    SecurityToken idpSecToken = getIdpSecurityToken();
+    if (idpSecToken != null) {
+      if (shouldAuditSubject(idpSecToken)) {
+        SecurityLogger.audit(
+            "Subject with admin privileges has logged out: {}",
+            new SecurityAssertionImpl(idpSecToken).getPrincipal().getName());
+      }
     }
   }
 
-  private boolean shouldAuditSubject() {
+  private boolean shouldAuditSubject(SecurityToken idpSecToken) {
```

Comment:
The original version of the method did not take an argument. Why the change?

Reply:
With the slight refactor, this leads to not having to call `getIdpSecurtyToken` again, when it already knows it from the only place it comes from and is being called (`logSecurityAuiditRole`)

Reply:
It gives the impression that different token can be passed in. If the same token (i.e. whatever `getIdpSecurityToken()` returns) is always what method wants, then it should be provided by the class as part of the class's internal state. If `getIdSecurityToken` is expensive in terms of resources it can be cached as a field on the object.

Reply:
This was a suggestion in a previous comment that made sense to me. Since it is only being called there and it is trying to use the same token, it seems reasonable to pass it in as a parameter.

Reply:
After further investigation, `logSecurityAuditSubject` and `shouldAuditSubject` were refactored out as helper methods for what looks like just to make the `logout()` method smaller and easier to follow. It previously used the same `securityToken` throughout the process. Unless there was a conscious effort to purposefully call and get that `securtyToken` multiple separate times, I see no issue in using it as a parameter.

Reply:
I had seen that the token is retrieved multiple times as well. Not sure if that was intentional, but I would guess that it was. I think this is fine as is. I don't think we want to be caching any security token because the design of this service is that it gets the user information as an input and logs them out. If we start saving user information as state, we run into concurrency problems.

Reply:
I agree with @brjeter's comment. The second we start storing and operating on state in this class, concurrency issues could pop up. I would argue that keeping the code as is is more robust.